app/src/main/java/org/wikipedia/page/PageFragmentLoadState.kt

```diff
@@ -116,80 +113,72 @@ class PageFragmentLoadState(private var model: PageViewModel,
         fragment.onPageLoadError(caught)
     }
 
-    private fun pageLoadCheckReadingLists() {
-        model.title?.let {
+    private fun pageLoad() {
+        model.title?.let { title ->
             fragment.lifecycleScope.launch(CoroutineExceptionHandler { _, throwable ->
                 L.e("Page details network error: ", throwable)
                 commonSectionFetchOnCatch(throwable)
             }) {
-                model.readingListPage = AppDatabase.instance.readingListPageDao().findPageInAnyList(it)
-                pageLoadFromNetwork(it)
-            }
-        }
-    }
-
-    private suspend fun pageLoadFromNetwork(title: PageTitle) {
-        fragment.updateQuickActionsAndMenuOptions()
-        if (!fragment.isAdded) {
-            return
-        }
```

Comment:
Do we still need this in the `pageLoad()`?

Reply:
This is exactly the benefit of using the `lifecycleScope` from the parent fragment or activity. If the fragment is destroyed, the `lifecycleScope` will automatically cancel any running coroutines. So, if we follow this pattern, we will never again need to check if a fragment is added, or if a View is attached to a window.

Reply:
Got it, it makes sense to me. Thanks!